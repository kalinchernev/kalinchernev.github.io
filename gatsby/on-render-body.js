'use strict';

const React = require('react');
const siteConfig = require('../config.js');

// eslint-disable-next-line import/no-webpack-loader-syntax, import/no-unresolved
const katexStylesheet = require('!css-loader!../static/css/katex/katex.min.css');

const onRenderBody = ({ setHeadComponents }) => {
  const { useKatex } = siteConfig;

  if (useKatex) {
    setHeadComponents([
      React.createElement('style', {
        key: 'katex-inline-stylesheet',
        dangerouslySetInnerHTML: { __html: katexStylesheet.toString() },
      }),
    ]);
  }
};

module.exports = onRenderBody;
