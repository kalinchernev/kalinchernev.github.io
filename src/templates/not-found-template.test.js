// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import NotFoundTemplate from './not-found-template';
import siteMetadata from '../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../types';

describe('NotFoundTemplate', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => render(siteMetadata),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(<NotFoundTemplate />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
