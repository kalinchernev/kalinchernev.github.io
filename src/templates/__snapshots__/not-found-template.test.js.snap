// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundTemplate renders correctly 1`] = `
<div
  className="layout"
>
  <div
    className="sidebar"
  >
    <div
      className="sidebar__inner"
    >
      <div
        className="author"
      >
        <a
          href="/"
        >
          <img
            alt="Test name"
            className="author__photo"
            height="75"
            src="/test.jpg"
            width="75"
          />
        </a>
        <h2
          className="author__title"
        >
          <a
            className="author__title-link"
            href="/"
          >
            Test name
          </a>
        </h2>
        <p
          className="author__subtitle"
        >
          Test bio
        </p>
      </div>
      <nav
        className="menu"
      >
        <ul
          className="menu__list"
        >
          <li
            className="menu__list-item"
          >
            <a
              className="menu__list-item-link"
              href="/test/1/"
            >
              Test label 1
            </a>
          </li>
          <li
            className="menu__list-item"
          >
            <a
              className="menu__list-item-link"
              href="/test/2/"
            >
              Test label 2
            </a>
          </li>
          <li
            className="menu__list-item"
          >
            <a
              className="menu__list-item-link"
              href="/test/3/"
            >
              Test label 3
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="contacts"
      >
        <ul
          className="contacts__list"
        >
          <li
            className="contacts__list-item"
          >
            <a
              className="contacts__list-item-link"
              href="https://www.twitter.com/#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                className="icon"
                viewBox="0 0 26 28"
              >
                <title>
                  twitter
                </title>
                <path
                  d="M25.312 6.375c-0.688 1-1.547 1.891-2.531 2.609 0.016 0.219 0.016 0.438 0.016 0.656 0 6.672-5.078 14.359-14.359 14.359-2.859 0-5.516-0.828-7.75-2.266 0.406 0.047 0.797 0.063 1.219 0.063 2.359 0 4.531-0.797 6.266-2.156-2.219-0.047-4.078-1.5-4.719-3.5 0.313 0.047 0.625 0.078 0.953 0.078 0.453 0 0.906-0.063 1.328-0.172-2.312-0.469-4.047-2.5-4.047-4.953v-0.063c0.672 0.375 1.453 0.609 2.281 0.641-1.359-0.906-2.25-2.453-2.25-4.203 0-0.938 0.25-1.797 0.688-2.547 2.484 3.062 6.219 5.063 10.406 5.281-0.078-0.375-0.125-0.766-0.125-1.156 0-2.781 2.25-5.047 5.047-5.047 1.453 0 2.766 0.609 3.687 1.594 1.141-0.219 2.234-0.641 3.203-1.219-0.375 1.172-1.172 2.156-2.219 2.781 1.016-0.109 2-0.391 2.906-0.781z"
                />
              </svg>
            </a>
          </li>
          <li
            className="contacts__list-item"
          >
            <a
              className="contacts__list-item-link"
              href="https://github.com/#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                className="icon"
                viewBox="0 0 26 28"
              >
                <title>
                  github
                </title>
                <path
                  d="M10 19c0 1.141-0.594 3-2 3s-2-1.859-2-3 0.594-3 2-3 2 1.859 2 3zM20 19c0 1.141-0.594 3-2 3s-2-1.859-2-3 0.594-3 2-3 2 1.859 2 3zM22.5 19c0-2.391-1.453-4.5-4-4.5-1.031 0-2.016 0.187-3.047 0.328-0.812 0.125-1.625 0.172-2.453 0.172s-1.641-0.047-2.453-0.172c-1.016-0.141-2.016-0.328-3.047-0.328-2.547 0-4 2.109-4 4.5 0 4.781 4.375 5.516 8.188 5.516h2.625c3.813 0 8.188-0.734 8.188-5.516zM26 16.25c0 1.734-0.172 3.578-0.953 5.172-2.063 4.172-7.734 4.578-11.797 4.578-4.125 0-10.141-0.359-12.281-4.578-0.797-1.578-0.969-3.437-0.969-5.172 0-2.281 0.625-4.438 2.125-6.188-0.281-0.859-0.422-1.766-0.422-2.656 0-1.172 0.266-2.344 0.797-3.406 2.469 0 4.047 1.078 5.922 2.547 1.578-0.375 3.203-0.547 4.828-0.547 1.469 0 2.953 0.156 4.375 0.5 1.859-1.453 3.437-2.5 5.875-2.5 0.531 1.062 0.797 2.234 0.797 3.406 0 0.891-0.141 1.781-0.422 2.625 1.5 1.766 2.125 3.938 2.125 6.219z"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <div
        className="copyright"
      >
        Test copyright
      </div>
    </div>
  </div>
  <div
    className="page"
  >
    <div
      className="page__inner"
    >
      <h1
        className="page__title"
      >
        NOT FOUND
      </h1>
      <div
        className="page__body"
      >
        <p>
          You just hit a route that doesn't exist... the sadness.
        </p>
      </div>
    </div>
  </div>
</div>
`;
